cmd/namespace/validate.go

```diff
@@ -2,28 +2,43 @@ package namespace
 
 import (
 	"bytes"
+	"context"
 	"fmt"
 	"io/ioutil"
 
 	"github.com/ory/keto/internal/driver/config"
+	"github.com/ory/keto/internal/namespace"
 
 	"github.com/ory/x/jsonschemax"
 
 	"github.com/ory/jsonschema/v3"
 	"github.com/ory/x/cmdx"
+	"github.com/ory/x/logrusx"
 	"github.com/segmentio/objconv/yaml"
 	"github.com/spf13/cobra"
-
-	"github.com/ory/keto/internal/namespace"
 )
 
 func NewValidateCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "validate <namespace.yml> [<namespace2.yml> ...]",
+		Use:   "validate <namespace.yml> [<namespace2.yml> ...] | validate <config.yaml>",
 		Args:  cobra.MinimumNArgs(1),
-		Short: "Validate namespace files",
-		Long:  "Validate one or more namespace yaml files and get human readable errors. Useful for debugging.",
+		Short: "Validate namespace definitions",
+		Long: `validate
+Validates namespace definitions. Parses namespace yaml files or a configuration
+file containing namespace defitions. Returns human readable errors. Useful for
+debugging.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
+			// check user passed a config file
+			if len(args) == 1 {
```

Comment:
Ah I missed that global flag, but I see it now digging through the code.

I tend to like code where an argument can only be a single type. Since this flag exists, I'd say, let's change the behavior to this style

```sh
### VALID
./keto namespace validate -c config.yaml config2.yaml

./keto namespace validate namespace.yaml ns2.yaml

# namespaces and configs are all validated
./keto namespace validate namespace.yaml ns2.yaml -c config.yaml

### INVALID

# interprets config.yaml as a namespace file
./keto namespace validate config.yaml
```

(the `args` are always treated as namespace files rather than doing the weird type-discovery as this code is now)

What do you think?

Reply:
Sounds perfect :+1: